chartfx-dataset/src/main/java/de/gsi/dataset/serializer/spi/BinarySerialiser.java

```diff
@@ -279,383 +313,398 @@ public static double getDouble(final IoBuffer readBuffer) {
         return null;
     }
 
-    public static String getEnumTypeList(final IoBuffer readBuffer) {
+    @Override
+    public String getEnumTypeList() {
         // read value vector
-        final String enumSimpleName = getString(readBuffer);
-        final String enumName = getString(readBuffer);
-        final String enumTypeList = getString(readBuffer);
-        final String enumState = getString(readBuffer);
-        final int enumOrdinal = getInteger(readBuffer);
+        buffer.getStringISO8859(); // enumSimpleName
+        buffer.getStringISO8859(); // enumName
+        final String enumTypeList = buffer.getStringISO8859();
+        buffer.getStringISO8859(); // enumState
+        buffer.getInt(); // enumOrdinal
 
         return enumTypeList;
     }
 
-    public static FieldHeader getFieldHeader(final IoBuffer readBuffer) {
-        final String fieldName = readBuffer.getString();
-        final byte dataTypeByte = readBuffer.getByte();
-        final DataType dataType = DataType.fromByte(dataTypeByte);
-
-        if (dataType.isScalar()) {
-            final long pos = readBuffer.position();
-            final long nBytesToRead = dataType == DataType.STRING ? readBuffer.getInt() + 4
-                                                                  : dataType.getPrimitiveSize();
-            readBuffer.position(pos);
-
-            return new FieldHeader(fieldName, dataType, new int[] { 1 }, readBuffer.position(), nBytesToRead);
+    @Override
+    public WireDataFieldDescription getFieldHeader() {
```

Comment:
Codacy found an issue: [The method 'getFieldHeader()' has an NPath complexity of 384, current threshold is 200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5905529)